cmd/podman/images/scp.go

```diff
@@ -0,0 +1,383 @@
+	if err != nil {
+		return errors.Wrapf(err, "could not make config")
+	}
+	abiEng, err := registry.NewImageEngine(cmd, args) // abi native engine
+	if err != nil {
+		return errors.Wrapf(err, "could not make abi engine")
+	}
+	url := ""
+	iden := ""
+	serv := map[string]config.Destination{}
+	cfg, err := config.ReadCustomConfig() // get ready to set ssh destination if necessary
+	if err != nil {
+		return err
+	}
+	conn := ""
+	if strings.Contains(args[0], "::") || len(args) == 2 { // if we have a connection name specified
+		if len(args) == 2 && !(strings.Contains(args[1], "::") && strings.Contains(args[0], "::")) { // if an image is specified, this mean we are loading from our client
+			conn = args[1]
+		} else { // else we are loading from the ssh/remote client
+			scpOpts.ToFrom = false
+			conn = args[0]
+		}
+		splitEnv := strings.SplitN(conn, "::", 2)
+		if len(splitEnv) == 2 {
+			conn = splitEnv[0]
+			if len(args) == 2 && !(strings.Contains(args[1], "::") && strings.Contains(args[0], "::")) && splitEnv[1] != "" { // if we have two arguments and its not remote -> remote
+				scpOpts.Load.Input = splitEnv[1]
+			} else {
+				scpOpts.ImageName = splitEnv[1]
+			}
+		}
+		if len(cfg.Engine.ServiceDestinations) == 0 { // no connections in list >:/
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
+
+		for name, data := range cfg.Engine.ServiceDestinations { // loop through all of the connections until we find outs
+			if name == conn {
+				iden = data.Identity
+				url = data.URI
+			}
+		}
+		if iden == "" && url == "" { // no match, warn user and do a manual connection.
+			url = "ssh://" + conn
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
+	} else { // else we default to remote or whatever the default connection is
+		connStr := cfg.Engine.ActiveService
+		if len(cfg.Engine.ServiceDestinations) == 0 {
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
+
+		for name, data := range cfg.Engine.ServiceDestinations {
+			if name == connStr {
+				iden = data.Identity
+				url = data.URI
+			}
+		}
+		dst := config.Destination{}
+		if iden != "" {
+			dst.URI = url
+			dst.Identity = iden
+		} else {
+			dst.URI = url
+		}
+		serv[connStr] = dst
+	}
+	optionsTunnel := entities.PodmanConfig{}
+	confR.Engine = config.EngineConfig{Remote: true, CgroupManager: "cgroupfs", ServiceDestinations: serv} // pass the service dest (either remote or something else) to engine
+	if iden != "" {
+		optionsTunnel = entities.PodmanConfig{Config: confR, EngineMode: entities.TunnelMode, Remote: true, Identity: iden, URI: url}
+	} else {
+		optionsTunnel = entities.PodmanConfig{Config: confR, EngineMode: entities.TunnelMode, Remote: true, URI: url}
+	}
```

Comment:
```go
optionsTunnel := { /* the common settings */ }
if iden != "" {
    optionsTunnel.Identity = iden
}
```
would be much clearer about what this does.

Reply:
ah yes, that makes much more sense thanks